ft-sdk/src/session/mod.rs

```diff
@@ -0,0 +1,143 @@
+mod session_data;
+
+pub use ft_sys_shared::TRACKER_KEY;
+pub use session_data::SessionData;
+
+#[derive(Clone, Debug)]
+pub struct SessionID(pub String);
+
+impl SessionID {
+    /// Create a new session entry with the given user ID.
+    /// If the user ID is None, the session will be created without a user ID.
+    pub fn create(
+        conn: &mut ft_sdk::Connection,
+        user_id: Option<ft_sdk::auth::UserId>,
+        data: Option<serde_json::Value>,
+    ) -> Result<Self, ft_sdk::Error> {
+        use diesel::prelude::*;
+        use ft_sdk::schema::fastn_session;
+
+        let session_id = ft_sdk::utils::uuid_v8();
+
+        let data = match data {
+            Some(d) => serde_json::to_string(&d)?,
+            None => "{}".to_string(),
+        };
+
+        let user_id = user_id.map(|u| u.0);
+
+        diesel::insert_into(fastn_session::table)
+            .values((
+                fastn_session::id.eq(&session_id),
+                fastn_session::uid.eq(user_id),
+                fastn_session::created_at.eq(ft_sdk::env::now()),
+                fastn_session::updated_at.eq(ft_sdk::env::now()),
+                fastn_session::data.eq(data),
+            ))
+            .execute(conn)?;
+
+        Ok(Self(session_id))
+    }
+
+    pub fn from_string<S: AsRef<str>>(s: S) -> Self {
+        Self(s.as_ref().to_string())
+    }
+
+    /// Set the user ID for the given session.
+    /// this also clears existing session data
```

Comment:
This is not good. We are planning to store dark mode preference in session, this preference should not get lost when logging in. When logging out we delete the session cookie, but not the session row in db (for debugging etc, and will be removed after a period by some background worker).

Reply:
If a visitor (anonymous) user tries to subscribe to a newsletter, we store its `subscription_uid` in the session (we have uid because we created a user that can't yet log in). When this same user goes to create an account (or login) with a different email I want the session data to be reset.

How do I handle this case then?

Reply:
Hmm. So the harm here is that user will not see subscription choice (because they are subscribed with another email id), even tho they are not subscribed with this email id. Can we handle this in subscription app explicitly instead of clearing the session? Like if `subscription_uid != session.uid` show the dialog?